executor/src/witgen/vm_processor.rs

```diff
@@ -118,8 +127,10 @@ impl<'a, 'b, 'c, T: FieldElement, Q: QueryCallback<T>> VmProcessor<'a, 'b, 'c, T
         Self { processor, ..self }
     }
 
-    pub fn finish(self) -> (FinalizableData<T>, DegreeType) {
-        (self.processor.finish(), self.degree)
+    /// Returns the updated data & publics, and the length of the block.
+    pub fn finish(self) -> (FinalizableData<T>, BTreeMap<&'a str, T>, DegreeType) {
```

Comment:
This return type is getting quite complicated.

Could the publics be added to `FinalizableData`?

Reply:
Added a new type for the block + publics